src/lib/components/ChatInput.svelte

```diff
@@ -53,73 +77,341 @@
         hasFocus = false;
       },
     });
+
   });
 
+  // Flag to track whether an image is being uploaded
+  let isUploading = $state(false);
+
+  // Flag to track whether a file is being dragged over the drop area
+  let isDragOver = $state(false);
+
+  // Consolidated image file processing logic
+  async function processImageFile(file: File, input?: HTMLInputElement) {
```

Comment:
Can't help but wonder if we should call `processImageFile` after the user submits the message and just show the image in the browser beforehand. I guess I'm worried about uploading a bunch of images and storing them that are never attached to a message. Probably fine for now but perhaps worth a comment to clean up "dangling images" later.

Reply:
> I guess I'm worried about uploading a bunch of images and storing them that are never attached to a message

I'm having some difficulty parsing this statement. do you mean, someone uploads an image and then in message input field they backspace and delete the image from the message?

Reply:
yes, exactly. So they never actually post it.

Reply:
the upside is if they do that, the image doesn't go anywhere except into their own pds. we don't store the images in our crdt

Reply:
Totally happy with this being a refactor later, or something we deal with in some other way. We could even have a cron that takes care of cleaning things up from time to time.

Reply:
if they upload an image and don't use it I don't see why we need to clean it up? if they really care that much can't they use something like https://pdsls.dev? Having a task on our end that manages their pds seems a little out of scope

Reply:
> We could render the image in the browser, locally, without uploading it, until they submit the message is what I was trying to say.

i understand now, i can refactor this for sure

Reply:
We were gonna store in CRDT but many reasons to use pds instead. Top two were csam moderation and CDN